### PR TITLE
Remove build docker volume if OS_BUILD_ENV_CLEAN_BUILD_VOLUME is set

### PR DESCRIPTION
**Pull request by ingvagabund**
SSIA

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Volumes are created by `hack/lib/build/environment.sh` [here](https://github.com/openshift/origin/blob/master/hack/lib/build/environment.sh#L29) and [here](https://github.com/openshift/origin/blob/master/hack/lib/build/environment.sh#L39) -- I think the most appropriate place for cleanup would be in `os::build::environment::cleanup` through a environment variable that would optionally clean these up

**Comment by ingvagabund** [comment]
@stevekuznetsov saying it should be called optionally (e.g. when `OS_BUILD_ENV_CLEAN_VOLUMES=true`) when `os::build::environment::cleanup` is run?

**Comment by stevekuznetsov** [comment]
@ingvagabund yes, that sounds googd

**Comment by ingvagabund** [comment]
@detiber FYI

**Comment by ingvagabund** [comment]
@stevekuznetsov PTAL

**Comment by detiber** [comment]
@ingvagabund lgtm

File: hack/lib/build/environment.sh [hunk 1]
```diff
@@ -99,15 +99,21 @@ readonly -f os::build::environment::release::workingdir
...
     os::log::debug "Removing container ${container}"
     docker rm "${container}" > /dev/null
 
     if [[ -z "${OS_BUILD_ENV_REUSE_TMP_VOLUME:-}" ]]; then
-      os::log::debug "Removing tmp volume"
-      os::build::environment::remove_volume "${OS_BUILD_ENV_TMP_VOLUME}"
+      os::log::debug "Removing tmp build volume"
+      os::build::environment::remove_volume "${tmp_volume}"
+    fi
+    if [[ -n ${OS_BUILD_ENV_CLEAN_BUILD_VOLUME:-} ]]; then
```
Comment (stevekuznetsov) [inline hack/lib/build/environment.sh]: quote this expansion

File: hack/lib/build/environment.sh [hunk 2]
```diff
@@ -99,15 +99,21 @@ readonly -f os::build::environment::release::workingdir
...
+      os::log::debug "Removing build volume"
+      os::build::environment::remove_volume ${volume}
```
Comment (stevekuznetsov) [inline hack/lib/build/environment.sh]: quote this expansion

File: hack/lib/build/environment.sh (hunk 1 above)
Comment (ingvagabund) [reply hack/lib/build/environment.sh]: fixed

File: hack/lib/build/environment.sh (hunk 2 above)
Comment (ingvagabund) [reply hack/lib/build/environment.sh]: fixed

**Pull request merged**